python/lsst/daf/butler/_location.py

```diff
@@ -49,7 +49,12 @@ class Location:
 
     __slots__ = ("_datastoreRootUri", "_path", "_uri")
 
-    def __init__(self, datastoreRootUri: None | ResourcePathExpression, path: ResourcePathExpression):
+    def __init__(
+        self,
+        datastoreRootUri: None | ResourcePathExpression,
+        path: ResourcePathExpression,
+        _trusted_path: bool = False,
```

Comment:
`Location` itself is already sufficiently internal that I don't think you need to make this argument extra-private.